tcl/db.tcl

```diff
@@ -387,7 +387,11 @@ proc qc::db_trans {args} {
 	    } else {
 		incr db_trans_level($db) -1
 	    }
-	    return -code $code $result
+            
+            if { $return_code == 2 } {
```

Comment:
If we wanted to support `return -code x` statements I think that updating this code to:

```tcl
if { $return_code == 2 && [dict get $options -code] == 0 } {
    ...
}
```

would allow us to do so. I'd need to test more cases though to make sure they'd all be covered. Not entirely sure if that support is necessary at the moment.